docs/src/content/en/docs/agents/input-processors.mdx

```diff
@@ -0,0 +1,438 @@
+---
+title: "Input Processors"
+description: "Learn how to use input processors to intercept and modify agent messages before they reach the language model."
+---
+
+# Input Processors
+
+Input processors are powerful middleware components to add guardrails to prompts. They allow you to intercept, modify, validate, or filter messages before they are sent to the language model. They provide a flexible way to implement pre-processing logic, content filtering, message transformation, and early termination of agent requests.
+
+## Overview
+
+Input processors run sequentially before the agent sends messages to the language model. Each processor receives a context object and must return an array of `MastraMessageV2` objects. The returned array is then passed as input to the next processor in the chain.
+
+The context object contains:
+- **`messages`**: An array of `MastraMessageV2` objects representing the current messages in the pipeline. You can mutate this array directly and return it.
+- **`abort(reason?)`**: A function to terminate processing early with an optional reason.
+
+This design allows you to build sophisticated and predictable message processing pipelines with validation, transformation, and conditional logic.
+
+## Basic Usage
+
+### Creating an Input Processor
+
+To create an input processor, define an object that conforms to the `InputProcessor` interface.
+
+```typescript
+import { Agent, type InputProcessor, type MastraMessageV2 } from "@mastra/core/agent";
+
+const contentFilter: InputProcessor = {
```

Comment:
We should call out that you can use a class too and specify when you'd do one or the other (inline vs reusable)